### PR TITLE
fix: simplify mobile layout tabs

### PR DESCRIPTION
**Pull request by moT01**
I was just trying to fix https://github.com/freeCodeCamp/freeCodeCamp/pull/44247#discussion_r763078223... and, well, ended up doing this.

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

**Comment by ojeytonwilliams** [comment]
@moT01 my thinking with `hasEditableBoundaries` was just to have a single value that determines if we need to show the jaws or not. What's the rationale for making that a property of the meta?  It seems to me that `hasEditableBoundaries` is really a challenge property rather than a block property and that this approach just expands our configs.

**Comment by moT01** [comment]
Yes, I should have left a better description in the PR - sorry.

> It seems to me that `hasEditableBoundaries` is really a challenge property

Yea, I agree. Kinda seems like `usesMultifileEditor` probably is, as well :wink: It was just easier to do it this way 😬

Well, I was looking into fixing the issue linked in the description. It didn't feel like the tab navigation/state belonged in redux, so I tried moving it to the component state. It worked fine, with one exception - a slight delay in calculating this `hasEditableBondaries` value, which itself is determined by the challengeFiles stored in redux, made it not go to the correct tab when navigating from another page on the initial render. Hard refresh worked. Not sure that's a great explanation. Anyway - making it part of the meta solved it, and seemed to speed things up a little.

As a side thought, this property might be valuable in other areas. Potentially, with the new cert projects or something. I'm open to other thoughts and suggestions. We don't have to go this route.

**Comment by ojeytonwilliams** [comment]
Yeah, I take that point. My thinking was just that all files in a given block should use the multifile editor, but not necessarily have editable regions.

Thing is, I *think* we'll never have projects and steps in the same block. So, yeah, `hasEditableRegion` is a block property after all.

Did you happen to spot any other unnecessary renders? It's possible I've made this mistake in other places.

**Comment by moT01** [comment]
> Did you happen to spot any other unnecessary renders?

No - I'm not sure it was an unnecessary render. It just didn't have the `hasEditableBoundaries` value set correctly initially. And the way it was getting set from the other redux property was just a little goofy. I'm not sure what the mistake was - having properties in redux that don't necessarily need to be there - or having some like cyclical setting of redux values? Doesn't matter - no, it looks good. I guess if we wanted to move other properties to the meta, it could speed things up?

File: client/src/templates/Challenges/classic/mobile-layout.tsx [hunk 1]
```diff
@@ -1,52 +1,47 @@
...
+}
+
+class MobileLayout extends Component<MobileLayoutProps, MobileLayoutState> {
   static displayName: string;
-  componentDidMount() {
-    if (this.props.currentTab !== 1) this.props.moveToTab(1);
-  }
+
+  state: MobileLayoutState = {
+    currentTab: this.props.hasEditableBoundaries ? 'editor' : 'instructions'
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/classic/mobile-layout.tsx:33]:
```suggestion
    currentTab: this.props.hasEditableBoundaries ? Tab.Editor : Tab.Instructions
```

File: client/src/templates/Challenges/classic/mobile-layout.tsx [hunk 2]
```diff
@@ -1,52 +1,47 @@
...
+  };
+
+  switchTab = (tab: string) => {
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/classic/mobile-layout.tsx:36]:
```suggestion
  switchTab = (tab: Tab) => {
```

File: client/src/templates/Challenges/classic/mobile-layout.tsx [hunk 3]
```diff
@@ -69,50 +64,48 @@ class MobileLayout extends Component<MobileLayoutProps> {
...
-          defaultActiveKey={1}
+          defaultActiveKey={currentTab}
           id='challenge-page-tabs'
-          onSelect={moveToTab}
+          onSelect={this.switchTab}
         >
           {!hasEditableBoundaries && (
             <TabPane
-              eventKey={eventKeys.shift()}
+              eventKey='instructions'
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/classic/mobile-layout.tsx:77]:
```suggestion
              eventKey={Tab.Instructions}
```

File: client/src/templates/Challenges/classic/mobile-layout.tsx [hunk 4]
```diff
@@ -69,50 +64,48 @@ class MobileLayout extends Component<MobileLayoutProps> {
...
               title={i18next.t('learn.editor-tabs.info')}
             >
               {instructions}
             </TabPane>
           )}
           <TabPane
-            eventKey={eventKeys.shift()}
+            eventKey='editor'
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/classic/mobile-layout.tsx:84]:
```suggestion
            eventKey={Tab.Editor}
```

File: client/src/templates/Challenges/classic/mobile-layout.tsx [hunk 5]
```diff
@@ -69,50 +64,48 @@ class MobileLayout extends Component<MobileLayoutProps> {
...
             title={i18next.t('learn.editor-tabs.code')}
             {...editorTabPaneProps}
           >
             {projectBasedChallenge && <EditorTabs />}
             {editor}
           </TabPane>
           <TabPane
-            eventKey={eventKeys.shift()}
+            eventKey='console'
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/classic/mobile-layout.tsx:92]:
```suggestion
            eventKey={Tab.Console}
```

File: client/src/templates/Challenges/classic/mobile-layout.tsx [hunk 6]
```diff
@@ -69,50 +64,48 @@ class MobileLayout extends Component<MobileLayoutProps> {
...
             title={i18next.t('learn.editor-tabs.tests')}
             {...editorTabPaneProps}
           >
             {testOutput}
           </TabPane>
           {hasNotes && projectBasedChallenge && (
             <TabPane
-              eventKey={eventKeys.shift()}
+              eventKey='notes'
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/classic/mobile-layout.tsx:100]:
```suggestion
              eventKey={Tab.Notes}
```

File: client/src/templates/Challenges/classic/mobile-layout.tsx [hunk 7]
```diff
@@ -69,50 +64,48 @@ class MobileLayout extends Component<MobileLayoutProps> {
...
               title={i18next.t('learn.editor-tabs.notes')}
             >
               {notes}
             </TabPane>
           )}
           {hasPreview && (
             <TabPane
-              eventKey={eventKeys.shift()}
+              eventKey='preview'
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/classic/mobile-layout.tsx:108]:
```suggestion
              eventKey={Tab.Preview}
```

**Review by ojeytonwilliams — DISMISSED** [review]
Looks nice, Tom.  We can use an enum for the event keys, if you like.  That feels a little DRYer and more type-safe to me.

It's fine as is, though.

**Review by ojeytonwilliams — DISMISSED** [review]
LGTM 👍

**Review by ojeytonwilliams — APPROVED** [review]

File: curriculum/getChallenges.js [hunk 8]
```diff
@@ -304,6 +305,7 @@ ${getFullPath('english')}
     usesMultifileEditor
   } = meta;
   challenge.block = dasherize(blockName);
+  challenge.hasEditableBoundaries = !!hasEditableBoundaries;
```
Comment (ShaunSHamilton) [inline curriculum/getChallenges.js:308]: Thank you, for this! Far too often, we used to end up with `undefined`s.

**Review by ShaunSHamilton — APPROVED** [review]
This LWTM 👍 

---

I am always a fan of removing stuffs, and _yay for enums_

**Pull request merged**